aion/web/elm/src/Update.elm

```diff
@@ -19,14 +19,30 @@ import Phoenix.Socket
 import Phoenix.Push
 import Task
 import Toasty
+import Multiselect
 import Socket exposing (initializeRoom, leaveRoom)
 
 
 update : Msg -> Model -> ( Model, Cmd Msg )
 update msg model =
     case msg of
         OnFetchRooms response ->
-            { model | rooms = response } ! []
+            let
+                newModel =
+                    { model | rooms = response }
+
+                categoryList =
+                    case newModel.rooms of
+                        RemoteData.Success roomsData ->
+                            List.map (\room -> ( toString (room.id), room.name )) roomsData.data
+
+                        _ ->
+                            []
+
+                oldPanelData =
+                    model.panelData
+            in
+                { model | panelData = { oldPanelData | categoryMultiSelect = Multiselect.initModel categoryList "id" } } ! []
```

Comment:
I think we can use the let part to actually assign the final value to a name like updatedCategoryMultiSelect (or whatever seems more suitable for you) and then update the model like:
```
{model | panelData = {oldPanelData | categoryMultiSelect = updatedCategoryMultiSelect}} 

```
Having function calls in the record updates makes it harder to read the code.